server/neptune/workflows/internal/deploy/revision/queue/pull_rebaser.go

```diff
@@ -0,0 +1,159 @@
+package queue
+
+import (
+	"context"
+	"path/filepath"
+	"strings"
+	"time"
+
+	"github.com/docker/docker/pkg/fileutils"
+	"github.com/pkg/errors"
+	key "github.com/runatlantis/atlantis/server/neptune/context"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/github"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/activities/terraform"
+	"github.com/runatlantis/atlantis/server/neptune/workflows/internal/config/logger"
+	"go.temporal.io/sdk/workflow"
+)
+
+// GH API Ratelimit resets every hour, 2 hours should be enough for the GH API call failures to autoresolve
+const RebaseGithubScheduleToCloseTimeout = 2 * time.Hour
+
+type githubRebaseActivities interface {
+	GithubListOpenPRs(ctx context.Context, request activities.ListOpenPRsRequest) (activities.ListOpenPRsResponse, error)
+	GithubListModifiedFiles(ctx context.Context, request activities.ListModifiedFilesRequest) (activities.ListModifiedFilesResponse, error)
+}
+
+type prRevisionSetterActivities interface {
+	SetPRRevision(ctx context.Context, request activities.SetPRRevisionRequest) (activities.SetPRRevisionResponse, error)
+}
+
+type PullRebaser struct {
+	GithubActivities           githubRebaseActivities
+	PrRevisionSetterActivities prRevisionSetterActivities
+}
+
+func (p *PullRebaser) RebaseOpenPRsForRoot(ctx workflow.Context, repo github.Repo, root terraform.Root) error {
+	originalOpts := workflow.GetActivityOptions(ctx)
+
+	// Configure github specific ScheduleToClose timeout for GH API call failures to autoresolve
+	// TODO: Add alarms on GithubListOpenPRs and GithubListModifiedFiles API call failures
+	ghOpts := originalOpts
+	ghOpts.ScheduleToCloseTimeout = RebaseGithubScheduleToCloseTimeout
```

Comment:
I meant to configure `ScheduleToCloseTimeout` such that we are able to auto-resolve GH API call failures when we hit our API ratelimit and add alarms on these activity execution failures. If it's not resolved within 2 hours, it's likely not a ratelimit issue. However, with the custom tq setup, we won't need to do this since we ratelimit the number of activities executed by the worker.